Error in csv_2_json while reading csv files

```
INFO:     2022-11-14 15:18:29,246 : 130.237.250.5:0 - "POST /csv_2_json HTTP/1.0" 500
[2022-11-14 15:18:29 +0100] [4] [ERROR] Exception in ASGI application
Traceback (most recent call last):
File "/home/worker/app/preClinVar/csv_parser.py", line 226, in csv_lines
next(csvreader)  # skip header
File "/usr/local/lib/python3.8/csv.py", line 111, in __next__
row = next(self.reader)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "/home/worker/venv/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 403, in run_asgi
result = await app(self.scope, self.receive, self.send)
File "/home/worker/venv/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
return await self.app(scope, receive, send)
File "/home/worker/venv/lib/python3.8/site-packages/fastapi/applications.py", line 269, in __call__
await super().__call__(scope, receive, send)
File "/home/worker/venv/lib/python3.8/site-packages/starlette/applications.py", line 124, in __call__
await self.middleware_stack(scope, receive, send)
File "/home/worker/venv/lib/python3.8/site-packages/starlette/middleware/errors.py", line 184, in __call__
raise exc
File "/home/worker/venv/lib/python3.8/site-packages/starlette/middleware/errors.py", line 162, in __call__
await self.app(scope, receive, _send)
File "/home/worker/venv/lib/python3.8/site-packages/starlette/exceptions.py", line 93, in __call__
raise exc
File "/home/worker/venv/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
await self.app(scope, receive, sender)
File "/home/worker/venv/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
raise e
File "/home/worker/venv/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
await self.app(scope, receive, send)
File "/home/worker/venv/lib/python3.8/site-packages/starlette/routing.py", line 670, in __call__
await route.handle(scope, receive, send)
File "/home/worker/venv/lib/python3.8/site-packages/starlette/routing.py", line 266, in handle
await self.app(scope, receive, send)
File "/home/worker/venv/lib/python3.8/site-packages/starlette/routing.py", line 65, in app
response = await func(request)
File "/home/worker/venv/lib/python3.8/site-packages/fastapi/routing.py", line 227, in app
raw_response = await run_endpoint_function(
File "/home/worker/venv/lib/python3.8/site-packages/fastapi/routing.py", line 160, in run_endpoint_function
return await dependant.call(**values)
File "/home/worker/app/preClinVar/main.py", line 93, in csv_2_json
file_lines = await csv_lines(file)
RuntimeError: coroutine raised StopIteration
[hiseq.clinical@cg-vm1 ~]$
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## [unreleased]
+### Fixed
+- Prevent CSV file parsing StopIteration error when provided files are empty
+
 ## [1.0.1]
 ### Fixed
 - Fixed Dockerfile building error resulting in prod image release failed to be pushed to Docker Hub

--- a/preClinVar/csv_parser.py
+++ b/preClinVar/csv_parser.py
@@ -221,9 +221,8 @@ async def csv_lines(csv_file):
         with file_copy as f:
             f.write(contents)
 
-        with open(file_copy.name, "r", encoding="utf-8") as csvf:
+        with open(file_copy.name) as csvf:
             csvreader = DictReader(csvf)
-            next(csvreader)  # skip header
             for row in csvreader:
                 lines.append(row)
 
@@ -231,4 +230,4 @@ async def csv_lines(csv_file):
         file_copy.close()  # Close temp file
         os.unlink(file_copy.name)  # Delete temp file
 
-    return lines
+    return lines[1:] if lines else lines  # skip the header

--- a/preClinVar/main.py
+++ b/preClinVar/main.py
@@ -100,7 +100,9 @@ async def csv_2_json(files: List[UploadFile] = File(...)):
     if not casedata_lines or not variants_lines:
         return JSONResponse(
             status_code=400,
-            content={"message": "Both 'Variant' and 'CaseData' csv files are required"},
+            content={
+                "message": "Both 'Variant' and 'CaseData' csv files are required and should not be empty"
+            },
         )
 
     # Convert lines extracted from csv files to a submission object (a dictionary)

--- a/tests/test_main.py
+++ b/tests/test_main.py
@@ -38,7 +38,10 @@ def test_csv_2_json_missing_file():
     response = client.post("/csv_2_json", files=files)
     # THEN the endpoint should return error
     assert response.status_code == 400
-    assert response.json()["message"] == "Both 'Variant' and 'CaseData' csv files are required"
+    assert (
+        response.json()["message"]
+        == "Both 'Variant' and 'CaseData' csv files are required and should not be empty"
+    )
 
 
 def test_csv_2_json_malformed_file():

